soundata/datasets/fsd50k.py

```diff
@@ -0,0 +1,478 @@
+
+    If you use the FSD50K Dataset please cite the following paper:
+
+    .. code-block:: latex
+        Eduardo Fonseca, Xavier Favory, Jordi Pons, Frederic Font, Xavier Serra. "FSD50K: an Open Dataset of
+        Human-Labeled Sound Events", arXiv:2010.00475, 2020.
+
+    The authors would like to thank everyone who contributed to FSD50K with annotations, and especially Mercedes
+    Collado, Ceren Can, Rachit Gupta, Javier Arredondo, Gary Avendano and Sara Fernandez for their commitment and
+    perseverance. The authors would also like to thank Daniel P.W. Ellis and Manoj Plakal from Google Research for
+    valuable discussions. This work is partially supported by the European Union’s Horizon 2020 research and innovation
+    programme under grant agreement No 688382 AudioCommons, and two Google Faculty Research Awards 2017 and 2018, and
+    the Maria de Maeztu Units of Excellence Programme (MDM-2015-0502).
+
+
+    License
+    -------
+
+    All audio clips in FSD50K are released under Creative Commons (CC) licenses. Each clip has its own license as
+    defined by the clip uploader in Freesound, some of them requiring attribution to their original authors and some
+    forbidding further commercial reuse. For attribution purposes and to facilitate attribution of these files to third
+    parties, we include a mapping from the audio clips to their corresponding licenses. The licenses are specified in
+    the files dev_clips_info_FSD50K.json and eval_clips_info_FSD50K.json. These licenses are CC0, CC-BY, CC-BY-NC and
+    CC Sampling+.
+8
+    In addition, FSD50K as a whole is the result of a curation process and it has an additional license: FSD50K is
+    released under CC-BY. This license is specified in the LICENSE-DATASET file downloaded with the FSD50K.doc zip file.
+
+    Usage of FSD50K for commercial purposes: If you'd like to use FSD50K for commercial purposes, please contact Eduardo
+    Fonseca and Frederic Font at eduardo.fonseca@upf.edu and frederic.font@upf.edu.
+
+
+    Feedback
+    --------
+
+    For further questions, please contact eduardo.fonseca@upf.edu, or join the freesound-annotator Google Group.
+
+"""
+
+import os
+from typing import BinaryIO, Optional, TextIO, Tuple
+
+import librosa
+import numpy as np
+import csv
+import json
+
+from soundata import download_utils
+from soundata import jams_utils
+from soundata import core
+from soundata import annotations
+from soundata import io
+
+
+BIBTEX = """
+TODO
+"""
+REMOTES = None
+
+LICENSE_INFO = "Creative Commons Attribution 4.0 International"
+
+
+class Clip(core.Clip):
+    """FSD50K Clip class
+
+    Args:
+        clip_id (str): id of the clip
+
+    Attributes:
+        audio_path (str): path to the audio file
+        clip_id (str): clip id
+        labels (soundata.annotation.Tags): tag (label) of the clip + confidence
+        sub_set (str): flag to identify if clip belongs to developement or evaluation set
```

Comment:
`sub_set` -> `split`

Reply:
Yes, that was my first idea! But then I realized that the `dev` split is also split between `train` and `val`, and the flag name from the ground-truth file is again `split`. How should we call the split between  `train` and `val`? `dev_split` or `sub_split` for instance?

Reply:
what about having `split` and returning `dev_train` and `dev_val`?

Reply:
I see. Sure, we can do it like that! I'll check how the implementation and workaround look like

Reply:
question - why not just call the field `split` and have it return `train`, `validation`, and `test` ?

Reply:
yeah that's better. I have implemented a single property `split`. Do you prefer `train`, `validation` and `test` or should we keep the FSD50K nomenclature which is `development`, `validation`, `evaluation`? I guess we need also to consider what is going to be the soundata standard nomenclature in this case.